### ISSUE TITLE
Easyshop's Country field does not support unicode characters

### ISSUE THREAD

**Issue by spanish** [ned14/Easyshop#23]
If I selected "España" (ñ=non ASCII character) I have this error:

Tipo de Excepción
    TypeError
Valor de la Excepción
    decomposition() argument 1 must be unicode, not str

Traceback (innermost last):

```
* Module ZPublisher.Publish, line 119, in publish
* Module ZPublisher.mapply, line 88, in mapply
* Module ZPublisher.Publish, line 42, in call_object
* Module zope.formlib.form, line 769, in __call__
* Module Products.PloneGetPaid._patch, line 44, in update
* Module zope.formlib.form, line 750, in update
* Module zope.formlib.form, line 594, in success
* Module easyshop.checkout.browser.address_add, line 32, in handle_next_action
* Module easyshop.checkout.browser.address_add, line 53, in createAndAdd
* Module plone.i18n.normalizer, line 73, in normalize
* Module plone.i18n.normalizer.base, line 71, in baseNormalize
```

TypeError: decomposition() argument 1 must be unicode, not str 

---

If I change "España" to "Espana" or "Spain", no problem.

Cordial greetings,

Manuel

**Comment by ned14** [comment]
Try u"España" and see what happens.

**Comment by spanish** [comment]
Now, no error, but in the dropdown, addresses blocks and pages, and in the emails, don't appears España. Appears u"España".

**Comment by ned14** [comment]
Oh I've just twigged what you meant - you meant you are typing España in as the delivery address or whatever, and Easyshop is spewing.

Ok, I've renamed the bug title to be clearer and thanks for the bug report.

Niall

**Comment by spanish** [comment]
Yes:

If I put España in countries list (EasyShop > Edit > Misc), I have the above error when I go to checkout page.

If I put u"España", no error, but I see u"España" everywhere.

Thanks and greetings,

Manuel

**Comment by petschki** [comment]
well ... i'm getting closer to git but not close enough :-)
this would fix the issue (easyshop.checkout.browser.address_add (line 54)

<pre>
if queryUtility(IIDNormalizer).normalize(safe_unicode(country)) == data.get("country"):
</pre>


and dont forget

<pre>
from Products.CMFPlone.utils import safe_unicode
</pre>

**Comment by ned14** [comment]
I see you've made your own fork - if you push updates to that then I can pull them from you into the Easyshop repo here.

Alternatively I've added you as a collaborator to this Easyshop too so I think it's just a question now for you to switch branches to this and push changes directly.

Slightly embarrassingly right now I don't have a working local Plone buildout! I tried upgrading my main workstation with a SSD and I tried to migrate the existing OS installs to the SSD. Needless to say, it absolutely hosed the lot :(. Best to completely reinstall from scratch I think, but I am waiting for some free time (need to install a new motherboard too, the old one doesn't like SSDs, probably is going to be my Christmas holidays at this rate!).

Niall
